singer-io/tap-mixpanel
# Tdl 14795 streams and sync should not be dict based

# Description of change
- Changed tap structure from dict based streams to class base.
- Updated error handling - removed nested try-catch
- Made bookmark key automatic
- Added end_date param - to handle so many requests
- Fixed pagination issue - Which was falling for child streams
- Added tap tester test cases - As mentioned in the description to test new code changes.

# Manual QA steps
- Older tap behavior should not be changed like state files and catalog files must remain the same.
 - Verify Error handling and backoffs are working properly.
 - Pagination issue got resolved so we should get all records for child streams.
 - Check bookmark keys must be automatic.
 - Tap-tester must be passed.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
